files/api/orchestrator.rb

```diff
@@ -9,16 +9,29 @@ def initialize(pe_console, username: nil, password: nil, token: nil, ssl_verify:
       @pe_client = PeEventForwarding::PeHttp.new(pe_console, port: 8143, username: username, password: password, token: token, ssl_verify: ssl_verify)
     end
 
-    def get_jobs(limit: nil, offset: nil, order: 'asc', order_by: 'name')
+    def get_jobs(offset: 0, order: 'asc', order_by: 'name', api_window_size: nil)
       params = {
-        limit:    limit,
+        limit:    api_window_size,
         offset:   offset,
         order:    order,
         order_by: order_by,
       }
-      response = pe_client.pe_get_request('orchestrator/v1/jobs', params)
+
+      api_window_size = api_window_size.to_i
+      response_items = []
+      response = ''
+      total_count = 0
+      loop do
+        response = pe_client.pe_get_request('orchestrator/v1/jobs', params)
+        response_body = JSON.parse(response.body)
+        total_count = response_body['pagination']['total']
+        response_items = response_items << response_body['items']
```

Comment:
All of the comments about the alignment and the `<<` operator apply here as well of course.